src/pkgcheck/checks/codingstyle.py

```diff
@@ -903,3 +903,75 @@ def feed(self, pkg):
                         yield RedundantDodir(
                             cmd.group('cmd'), line=dodir.group('call'),
                             lineno=lineno - 1, pkg=pkg)
+
+
+class UnquotedVariable(results.LineResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class UnquotedVariableCheck(Check):
+    """Scan ebuild for variables that should be quoted like D, FILESDIR, etc."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([UnquotedVariable])
+
+    message_commands = [
+        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
+    ]
+    var_names = [
+        "D", "DISTDIR", "FILESDIR", "S", "T", "ROOT", "BROOT", "WORKDIR",
+        # EAPI=3/Prefix vars
+        "ED", "EPREFIX", "EROOT",
+	    # variables for games.eclass
+		"Ddir", "GAMES_PREFIX_OPT", "GAMES_DATADIR",
+		"GAMES_DATADIR_BASE", "GAMES_SYSCONFDIR", "GAMES_STATEDIR",
+		"GAMES_LOGDIR", "GAMES_BINDIR",
+	    # variables for multibuild.eclass
+	    "BUILD_DIR",
+    ]
```

Comment:
Let's create a tuple here, just for extra immutability
```suggestion
    message_commands = (
        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
    )
    var_names = (
        "D", "DISTDIR", "FILESDIR", "S", "T", "ROOT", "BROOT", "WORKDIR",
        # EAPI=3/Prefix vars
        "ED", "EPREFIX", "EROOT",
	    # variables for games.eclass
		"Ddir", "GAMES_PREFIX_OPT", "GAMES_DATADIR",
		"GAMES_DATADIR_BASE", "GAMES_SYSCONFDIR", "GAMES_STATEDIR",
		"GAMES_LOGDIR", "GAMES_BINDIR",
	    # variables for multibuild.eclass
	    "BUILD_DIR",
    )
```

Reply:
Looks like I should also fix some indentation here...

Reply:
done